Refactor: Remove old sendMessage/showTitle

KingOfSquares opened the pull request:
- Moves away from old methods, properly references the console, fixes a couple of sound names

Next commit: Replacing some `StringBuilder`s with Components


Signed-off-by: KingSimon <19822231+KingOfSquares@users.noreply.github.com>

KingOfSquares committed:
Start moving away from old PGM Audience class
- MatchPlayer(State)/Party/Match etc. now implements adventure ForwardingAudience(creates an audience at creation) instead of going the detour through the old Audience class. This also results in a move to adventure objects for Titles/Components/Bossbars etc.
- Make a centralized AudienceProvider in PGMAudiences that can be passed around
- Move the widely used sendWarning to PGMAudiences as a static method

- Fixed a couple of sound names

Signed-off-by: KingSimon <19822231+KingOfSquares@users.noreply.github.com>

applenick commented on core/src/main/java/tc/oc/pgm/command/ClassCommand.java:
```diff
@@ -65,10 +65,12 @@ public void classList(Match match, MatchPlayer player) {
-            "-",
-            ChatColor.RED.toString()));
+        text(
+            LegacyFormatUtils.dashedChatMessage(
```
Switch this to use [TextFormatter#horizontalLineHeading](https://github.com/PGMDev/PGM/blob/dev/util/src/main/java/tc/oc/pgm/util/text/TextFormatter.java#L110) 😄

KingOfSquares replied:
Yeah I know, I just thought it would make sense to do such a big refactor step by step.

I was thinking TextFormatter should be refactored to use `MatchPlayer`s instead anyways

applenick replied:
Ah just be careful about that, since if an outside plugin wants to use the utils you don’t want them to be forced to use PGM just to get the MatchPlayer.

KingOfSquares replied:
Right, then the bukkit `Player` would also work, yeah? `player.getBukkit()`

applenick replied:
It depends on the method. For example I use the horizontalLineHeading methods in my requests plugin [here](https://github.com/applenick/Requests/blob/master/src/main/java/tc/oc/occ/Requests/RequestCommand.java#L245). If it were switched to player, then would no longer be able to run the command from the console.

KingOfSquares replied:
Then what about a `Locale` ? This seems to be the only need for the sender

applenick replied:
I don’t have time atm to look into what the sender is needed for, but awesome yeah if it’s something more general like `Locale` that would be the best path to take 👍

KingOfSquares committed:
rename console refrence

Signed-off-by: KingSimon <19822231+KingOfSquares@users.noreply.github.com>

Electroid commented on core/src/main/java/tc/oc/pgm/PGMAudiences.java:
```diff
@@ -0,0 +1,30 @@
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.plugin.Plugin;
+
+public class PGMAudiences {
```
Instead of having something like `PGMAudiences`, I would like this to live in `util`. 

`tc.oc.util.text.Audience` would be both an interface, with things like `sendWarning` wrapper, and 1-stop-shop to create audiences from players or groups of players.

KingOfSquares replied:
I see, any reason you decided to deprecate the provider that was in there?

The way im understanding this is that the provider also should stay in this Audience class

KingOfSquares committed:
Move back to PGM Audience
- Remove all references to adventure Audience in favour of our Audience
- remove deprecation
- add identities to ChatDispatcher

Signed-off-by: KingSimon <19822231+KingOfSquares@users.noreply.github.com>

KingOfSquares commented:
Thanks for taking my contribution into review Electroid. I hope this is closer to what you imagined

Electroid commented:
Great progress, willing to merge this in.
